JASP-Tests/R/tests/testthat/test-networkanalysis.R

```diff
@@ -0,0 +1,43 @@
+context("Network Analysis")
+
+# does not test
+# - error handling
+
+test_that("All tables results match", {
```

Comment:
I'd advise making these into separate tests, but if you want to group them all you should add labels (see below). If you do not do this and a test fails, you will not know which table failed.

Reply:
The reason for one test (with multiple checks) is that running a single network analysis can take some time. And as the tests are not adaptive to run for modified code only, I didn't want to suddenly add > 30 minutes of network tests that are not relevant to anybody but me. If this worry is unnecessary, let me know and I'll separate them.

Reply:
I don't think it matters terribly much, it just might be a tad clearer.
Note that the tests can be treated as a normal script.
So this would be possible:

```
context("Network Analysis")

# does not test
# - error handling

options <- JASPTools::analysisOptions("NetworkAnalysis")
options$variables <- c("contNormal", "contcor1", "contcor2")
options$tableCentrality <- TRUE
options$tableClustering <- TRUE
options$tableWeightsMatrix <- TRUE
options$tableLayout <- TRUE
results <- JASPTools::run("NetworkAnalysis", "debug.csv", options, view=FALSE, quiet=TRUE, sideEffects = "pkgLoading")

test_that("General TB table results match", {
  table <- results[["results"]][["generalTB"]][["data"]]
  expect_equal_tables(table,
    list("Network", 3, "2 / 3", 0.333333333333333))
})

test_that("Centrality TB table results match", {
  table <- results[["results"]][["centralityTB"]][["data"]]
  expect_equal_tables(table,
    list("contNormal", -0.577350269189626, -1.12003079401266, -1.14291478283658,
      "contcor1", 1.15470053837925, 0.803219560925303, 0.713968266021615,
      "contcor2", -0.577350269189626, 0.316811233087358, 0.428946516814968))
})
```

Reply:
Okay sounds good, the commit "Separated big unit test into multiple tests" implemented this.